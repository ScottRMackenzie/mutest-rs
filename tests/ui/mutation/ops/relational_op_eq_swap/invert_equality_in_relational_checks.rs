//@ print-mutations
//@ build
//@ stdout
//@ stderr: empty
//@ mutation-operators: relational_op_eq_swap

fn f() {
    let not_a_bool = 5 + 7 - 3;

    let mut tainted = false;
    for i in 0..10 {
        if i > 9 {
            tainted = true;
        }
    }

    let complex_bool = (not_a_bool + 1) == 10 || not_a_bool * 0 >= 1;

    if tainted && not_a_bool <= 9 {
        println!("tainted && not_a_bool <= 9");
    } else if tainted && not_a_bool > 9 {
        println!("tainted && not_a_bool > 9");
    } else if !(complex_bool as u8 >= 1) {
        println!("!complex_bool");
    } else {
        println!("something else");
    }

    match () {
        _ if complex_bool => { println!("complex_bool"); }
        () if (not_a_bool <= 9 && not_a_bool >= 9) || tainted => { println!("not_a_bool == 9 || tainted"); }
        _ => { println!("something else"); }
    }
}

#[test]
fn test() {
    f();
}
