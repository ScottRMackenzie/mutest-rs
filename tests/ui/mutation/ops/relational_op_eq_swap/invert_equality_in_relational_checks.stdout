[relational_op_eq_swap] add equality to relational operator `>` in f at tests/ui/mutation/ops/relational_op_eq_swap/invert_equality_in_relational_checks.rs:12:12: 12:17
  <-(0)- test

[relational_op_eq_swap] remove equality from relational operator `>=` in f at tests/ui/mutation/ops/relational_op_eq_swap/invert_equality_in_relational_checks.rs:17:50: 17:69
  <-(0)- test

[relational_op_eq_swap] remove equality from relational operator `<=` in f at tests/ui/mutation/ops/relational_op_eq_swap/invert_equality_in_relational_checks.rs:19:19: 19:34
  <-(0)- test

[relational_op_eq_swap] add equality to relational operator `>` in f at tests/ui/mutation/ops/relational_op_eq_swap/invert_equality_in_relational_checks.rs:21:26: 21:40
  <-(0)- test

[relational_op_eq_swap] remove equality from relational operator `>=` in f at tests/ui/mutation/ops/relational_op_eq_swap/invert_equality_in_relational_checks.rs:23:17: 23:40
  <-(0)- test

[relational_op_eq_swap] remove equality from relational operator `<=` in f at tests/ui/mutation/ops/relational_op_eq_swap/invert_equality_in_relational_checks.rs:31:16: 31:31
  <-(0)- test

[relational_op_eq_swap] remove equality from relational operator `>=` in f at tests/ui/mutation/ops/relational_op_eq_swap/invert_equality_in_relational_checks.rs:31:35: 31:50
  <-(0)- test

7 mutations; 7 safe; 0 unsafe (0 tainted)
